example/calendar.cpp

```diff
@@ -0,0 +1,165 @@
+constexpr auto day_str_size = day_pad_size + 2UL + day_pad_size;
+constexpr auto week_str_size = 7UL * day_str_size;
+constexpr auto max_weeks_in_month = 6;
+constexpr auto col_sep = "  ";
+constexpr auto row_sep = ' ';
+
+auto dates(ymd from, ymd to) {
+    return flux::iota(sys_days{from}, sys_days{to});
+}
+
+auto month_num = [](sys_days d1, sys_days d2) { 
+    return ymd(d1).month() == ymd(d2).month(); 
+};
+
+auto week_num = [](sys_days d1, sys_days d2) { 
+    return weekday(d1).iso_encoding() < weekday(d2).iso_encoding();
+};
+
+auto day_to_string = [](sys_days d) { 
+    auto day_num = static_cast<unsigned>(ymd(d).day());
+    auto day_str = (day_num < 10 ? " "s : ""s) + std::to_string(day_num);
+    std::string pad_str(day_pad_size, ' ');
+    return pad_str + day_str + pad_str;
+};
+
+auto month_name(sys_days d) { 
+    constexpr std::array months = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", 
+                                   "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+    auto year = std::to_string(static_cast<int>(ymd{d}.year()));
+    auto month = std::string{months[static_cast<unsigned>(ymd{d}.month()) - 1]};
+    auto title = month + '-' + year;
+    auto pad_size = (week_str_size - title.size()) / 2;
+    auto aligned_title = (std::string(pad_size, ' ') + title);
+    aligned_title.resize(week_str_size, ' ');
+    return aligned_title;
+};
+
+// In: sequence<sys_day> (week days)
+// Out: std::string (week string)
+auto week_to_string = [](flux::sequence auto&& week) { 
+    bool first_week_in_month = ymd(*flux::front(week)).day() == 1d;
+    auto week_str = FLUX_FWD(week)
+                     .map(day_to_string)
+                     .flatten()
+                     .template to<std::string>();                     
+    if (week_str.size() < week_str_size) {
+        std::string padding(week_str_size - week_str.size(), ' ');
+        week_str = first_week_in_month ? padding + week_str : week_str + padding;
+    }
+
+    return week_str;
+};
+
+// In: sequence<sys_day> (month days)
+// Out: sequence<std::string> (month calendar)
+auto to_week_lines = [](flux::sequence auto&& month) { 
+    auto month_str = month_name(*flux::front(month));
+    auto week_lines = FLUX_FWD(month)
+                        .chunk_by(week_num)
+                        .map(week_to_string);
+    std::vector<std::string> pad_lines(max_weeks_in_month - flux::count(week_lines), std::string(week_str_size, ' '));
+    return flux::chain(flux::single(std::move(month_str)), 
+                       std::move(week_lines), 
+                       std::move(pad_lines),
+                       flux::single(std::string(week_str_size, row_sep)));
+};
+
+auto append_column = [](std::vector<std::string>&& months_per_line, flux::sequence auto&& month) {
+    return flux::zip(flux::from(std::move(months_per_line)), FLUX_FWD(month).template to<std::vector>())
+        .map([](const auto& v) { return std::get<0>(v) + std::get<1>(v) + col_sep; })
```

Comment:
`zip` with two arguments returns a prvalue pair of references, so you could say 

```suggestion
        .map([](auto p) { return p.first + p.second + col_sep; })
```

which looks a bit nicer than `std::get<N>`

Reply:
Good point. Fixed!